### PR TITLE
AWS Lambda automate updating policies

### PR DESCRIPTION
**Pull request by RyPeck**
Using AWS Lambda, configured via the AWS CDK.

Python `dulwich` module to perform git operations. `get_aws_managed_policies.py` still works as before.

See `app.py` doc string for details on setting up and configuring.

### CODE REVIEW

**Comment by RyPeck** [comment]
Note - this depends on having an GitHub Account with a Personal Access token. I took the route of creating a "machine user" @RyPeck-bot to automate. Currently pointed at https://github.com/RyPeck/aws_managed_policies/tree/master and running every hour.

Next steps could be to get it working with SSH (there are Lambda Layers that make this look possible) but I thought I'd leave off here for now.

**Comment by z0ph** [comment]
Well done @RyPeck !!

**Comment by RyPeck** [comment]
Here is what an update looks like right now - https://github.com/RyPeck/aws_managed_policies/commit/f21818a061bcdadefe6e6f7559b7b572fff00d1f

File: auto_update_lambda/app.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+GH_BOT_REPO_OWNER - GitHub Repo Owner - Username or Org
+GH_BOT_COMMITTER  - Author/Committer String for Git
+
+2) Build the Lambda Deployment Package for the Lambda Code.
+
+```
+mkdir lambda_package; cd lambda_package/
+cp ../get_aws_managed_policies.py ../lambda-handler.py .
+pip3 install dulwich -t .
+zip -r ../deploymentPackage.zip .
```
Comment (0xdabbad00) [inline auto_update_lambda/app.py:24]: I think you could have avoided doing the manual zip here, but that's fine.

File: auto_update_lambda/app.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
+"""
```
Comment (0xdabbad00) [inline auto_update_lambda/app.py:1]: Awesome to see you went with the CDK!

Comment (0xdabbad00) [reply auto_update_lambda/app.py:1]: Should mention in the README how to deploy as a CDK app.

File: auto_update_lambda/lambda-handler.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            secret = get_secret_value_response["SecretString"]
+
+    return secret
+
+
+def update_repo(
+    github_bot_username,
+    gh_access_token,
+    repo_owner,
+    repo_path="/tmp/aws_managed_policies",
```
Comment (0xdabbad00) [inline auto_update_lambda/lambda-handler.py:62]: Nice job using the /tmp cache and later checking if this exists already or not!

**Comment by 0xdabbad00** [comment]
Looks like the bot is working against your repo. 👍 I did a review and this all looks good.  I want to actually deploy this before I merge it in, but hope to do that this weekend. Awesome work!

**Pull request closed without merging**